compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1023,7 +1038,14 @@ public void visit(BLangStreamType streamTypeNode) {
         BType streamType = new BStreamType(TypeTags.STREAM, constraintType, error, null);
         BTypeSymbol typeSymbol = type.tsymbol;
         streamType.tsymbol = Symbols.createTypeSymbol(typeSymbol.tag, typeSymbol.flags, typeSymbol.name,
-                typeSymbol.pkgID, streamType, typeSymbol.owner);
+                                                      typeSymbol.pkgID, streamType, typeSymbol.owner);
+
+        if (error != null) {
+            markParameterizedType(streamType, Arrays.asList(constraintType, error));
```

Comment:
we can call `markParameterizedType(streamType, constraintType)` always and only call for `error` conditionally. 

```Java
if (error != null) {
	markParameterizedType(streamType, error);
}
```